gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -276,24 +276,32 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
           if (!enableColumnarBatchScan) {
             TransformHints.tagNotTransformable(plan)
           } else {
-            val transformer = new BatchScanExecTransformer(plan.output, plan.scan,
-              plan.runtimeFilters)
-            TransformHints.tag(plan, transformer.doValidate().toTransformHint)
+            if (plan.runtimeFilters.nonEmpty) {
+              TransformHints.tagTransformable(plan)
+            } else {
+              val transformer = new BatchScanExecTransformer(plan.output, plan.scan,
+                plan.runtimeFilters)
+              TransformHints.tag(plan, transformer.doValidate().toTransformHint)
+            }
           }
         case plan: FileSourceScanExec =>
           if (!enableColumnarFileScan) {
             TransformHints.tagNotTransformable(plan)
           } else {
-            val transformer = new FileSourceScanExecTransformer(plan.relation,
-              plan.output,
-              plan.requiredSchema,
-              plan.partitionFilters,
-              plan.optionalBucketSet,
-              plan.optionalNumCoalescedBuckets,
-              plan.dataFilters,
-              plan.tableIdentifier,
-              plan.disableBucketedScan)
-            TransformHints.tag(plan, transformer.doValidate().toTransformHint)
+            if (plan.partitionFilters.nonEmpty) {
+              TransformHints.tagTransformable(plan)
```

Comment:
Also, looks some background is needed as notes. Same for above.

Reply:
Done